src/Aspire.Hosting/ApplicationModel/ResourceUrlsCallbackContext.cs

```diff
@@ -20,6 +20,19 @@ public class ResourceUrlsCallbackContext(DistributedApplicationExecutionContext
     /// </summary>
     public IResource Resource { get; } = resource;
 
+    /// <summary>
+    /// Gets an endpoint reference from <see cref="Resource"/> for the specified endpoint name.<br/>
+    /// If <see cref="Resource"/> does not implement <see cref="IResourceWithEndpoints"/> then returns <c>null</c>.
+    /// </summary>
+    /// <param name="name"></param>
+    /// <returns></returns>
+    public EndpointReference? GetEndpoint(string name) =>
+        Resource switch
+        {
+            IResourceWithEndpoints resourceWithEndpoints => resourceWithEndpoints.GetEndpoint(name),
+            _ => null
+        };
```

Comment:
Makes it easy for callers of `WithUrls` to get an endpoint on the resource without needing to do type checking/casting to `IResourceWithEndpoints`.

Reply:
Should it throw instead of returning null?

Reply:
Throwing is a bit mean how do you avoid throwing?